### PR TITLE
Fix @spec for `retry`

### PR DESCRIPTION
**Pull request by marcandre**

### CODE REVIEW

File: lib/wallaby/browser.ex [hunk 1]
```diff
@@ -141,7 +141,7 @@ defmodule Wallaby.Browser do
   loop forever (or until the test is killed by exunit).
   """
   @type sync_result :: {:ok, any()} | {:error, any()}
-  @spec retry((() -> sync_result), timeout) :: sync_result()
+  @spec retry((() -> sync_result), integer()) :: sync_result()
```
Comment (michallepicki) [inline lib/wallaby/browser.ex:144]:
```suggestion
  @spec retry((() -> sync_result), non_negative_integer()) :: sync_result()
```

**Review by michallepicki — APPROVED** [review]
Hi @marcandre !

I checked the reasoning behind this, for others: `timeout()` is one of [built-in types](https://hexdocs.pm/elixir/typespecs.html#built-in-types) and expands to `:infinity | non_neg_integer()`. This value is used in this private function:

```elixir
  defp max_time_exceeded?(start_time) do
    current_time() - start_time > max_wait_time()
  end
```

where it would not be valid to subtract `:infinity`, so it causes an `ArithmeticError` at runtime.

Thanks for the PR!

File: lib/wallaby/browser.ex [hunk 2]
```diff
@@ -141,7 +141,7 @@ defmodule Wallaby.Browser do
   loop forever (or until the test is killed by exunit).
   """
   @type sync_result :: {:ok, any()} | {:error, any()}
-  @spec retry((() -> sync_result), timeout) :: sync_result()
+  @spec retry((() -> sync_result), non_negative_integer()) :: sync_result()
```
Comment (michallepicki) [inline lib/wallaby/browser.ex:144]:
```suggestion
  @spec retry((() -> sync_result), non_neg_integer()) :: sync_result()
```

**Comment by marcandre** [comment]
Thanks!

**Pull request merged**